code/script.js

```diff
@@ -30,18 +49,177 @@ const showMessage = (message, sender) => {
+    //Here the user can select a breed from a dropdown menu based on their answer from Question 2
+    if (type === 'Small') {
+      inputWrapper.innerHTML = `
+        <select id='select'>
+          <option value="">Select Here</option>
+          <option value="Yorkshire Terrier">Yorkshire Terrier</option>
+          <option value="Chihuahua">Chihauhua</option>
+          <option value="Beagle">Beagle</option>
+          <option value="Pomeranian">Pomeranian</option>
+          <option value="Welsh Cardigan/Pembroke Corgi">Welsh Cardigan or Pembroke Corgi (Yes there 2 diff. breeds)</option>
+          <option value="Dachshund the sausage doggo">Dachshund aka Sausage dog!</option>
+        </select>`
+    } else if (type === 'Medium'){
+      inputWrapper.innerHTML = `
+        <select id='select'>
+          <option value="">Select Here</option>
+          <option value="Soft-Coated Wheaton Terrier">Soft-Coated Wheaton Terrier</option>
+          <option value="Siberian Husky">Siberian husky</option>
+          <option value="Basset Hound">Basset Hound</option>
+          <option value="Poodle">Poodle</option>
+          <option value="Golden Retriever">Golden Retriever</option>
+          <option value="Bull Terrier">Bull Terrier</option>
+        </select>`
+    } else {
+      inputWrapper.innerHTML = `
+        <select id='select'>
+          <option value="">Select Here</option>
+          <option value="Great Dane">Great Dane</option>
+          <option value="Saint Bernard">Saint Bernard</option>
+          <option value="American PitBull">American PitBull</option>
+          <option value="German Shepard">German Shepard</option>
+          <option value="Mastiff">Mastiff</option>
+          <option value="Irish Wolfhound">Irish Wolfhound</option>
+        </select>`
+    }
+    /*Defining 'select here, the EventListner 'change' is specific for <select> options
+    It tells the bot "Hey, we are changing because we've selected THIS option", similar to how
+    'click' means when something is clicked we change. */
+    const select = 
+      document
+        .getElementById('select');
+      select
+        .addEventListener('change', () => nextQuestion(select.value));
+  }
+
+  //This is Question 4
+  const giftWrapDog = () => {
+    indexDoggos++
+    botReply(`Ooo a ${select.value}? Great choice!`);
+    setTimeout(() => //This adds a 1 second delay before the bots chat appears.
+    botReply(`Would you like us to gift wrap your doggo?`), 2500);
+    
+    //we want the bot to remember the breed chosen before
+      inputWrapper.innerHTML = `
+        <button id="yes-pls-btn">🎁Yes, please.</button>
+        <button id="wtf-btn">Wtf... no!!</button>
+      ` 
+      document
+        .getElementById('yes-pls-btn')
+        .addEventListener('click', () => nextQuestion('Yes, please.'))
+      document
+        .getElementById('wtf-btn')
+        .addEventListener('click', () => nextQuestion('Wtf... no!'))
+  }
```

Comment:
This code structure is repeated, what is the ideal situation to create a more generic function to handle this task every time 🤓 Please try to do it 😸 If you will you can always ask me for more tips 😸